addons/account/models/ir_module.py

```diff
@@ -87,3 +87,13 @@ def _register_hook(self):
                 self.env.company,
             )
             del self.env.registry._auto_install_template
+
+    def module_uninstall(self):
+        # set `company.chart_template` to False for templates being uninstalled
+        modules_to_remove = self.mapped('name')
+        if any(module_to_remove.startswith('l10n') for module_to_remove in modules_to_remove):
+            unlinked_templates = [code for template in self.mapped('account_templates') for code in template]
+            self.env['res.company'].search([
+                    ('chart_template', 'in', unlinked_templates),
+                    ]).chart_template = False
```

Comment:
With code like this, you actually don't need the above comment anymore because it is easy to read now.
A comment could be put to explain why this is needed[^1] though (but I think the commit message is enough)
```suggestion
        unlinked_templates = [code for template in self.mapped('account_templates') for code in template]
        self.env['res.company'].search([
            ('chart_template', 'in', unlinked_templates),
        ]).chart_template = False
```

[^1]: it is needed because without removing the template, we might try to reload it and doing so can do `write`s on fields that have been remove with the unlinked module

Reply:
Done.